### PR TITLE
React 17 update

### PR DESCRIPTION
**Pull request by lieut-data**
## Summary
Thanks, @nevyangelova for kickstarting this! I pushed an extra change to update our `.nvmrc` to match `mattermost-webapp`, and then applied the integrity check removals that break M1 builds.

## Ticket Link
None.

### CODE REVIEW

**Comment by nevyangelova** [comment]
Thank you so much @lieut-data ! Perhaps run `node skip_integrity_check.js` and see if that updates the package-lock.

**Review by calebroseland — COMMENTED** [review]
Some externals may have to change before we can upgrade mattermost-webapp. Comment here: https://community.mattermost.com/core/pl/gqrq3kdk1pyw3ghydn5h8eoorh

**Comment by lieut-data** [comment]
Waiting on https://github.com/mattermost/mattermost-webapp/pull/11709 to see if we can resolve next steps.

File: .circleci/config.yml [hunk 1]
```diff
@@ -184,11 +184,6 @@ jobs:
       - *restore_go_cache
       - npm-dependencies
       - npm-e2e-dependencies
-      # Base image already bundle a golanci-lint but it's not updated enough (staticcheck linter needed)
```
Comment (lieut-data) [inline .circleci/config.yml:187]: Skip this since it's built into the pipeline already.

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-16.4.0
+16.10.0
```
Comment (lieut-data) [inline .nvmrc:1]: Match webapp.

File: Makefile [hunk 3]
```diff
@@ -108,7 +106,7 @@ endif
 webapp/node_modules: $(wildcard webapp/package.json)
 ifneq ($(HAS_WEBAPP),)
 	cd webapp && node skip_integrity_check.js
-	cd webapp && $(NPM) install
+	cd webapp && $(NPM) install --ignore-scripts
```
Comment (lieut-data) [inline Makefile:109]: CircleCI already did this, and seemed to be needed with mattermost-webapp updates.

File: webapp/src/components/backstage/backstage.tsx [hunk 4]
```diff
@@ -6,8 +6,7 @@ import {useLocation, useRouteMatch, matchPath} from 'react-router-dom';
 import {useSelector} from 'react-redux';
 import styled, {css} from 'styled-components';
 import {GlobalState} from '@mattermost/types/store';
-import {Theme} from 'mattermost-redux/types/themes';
-import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+import {getTheme, Theme} from 'mattermost-redux/selectors/entities/preferences';
```
Comment (lieut-data) [inline webapp/src/components/backstage/backstage.tsx:9]: Updated import.

File: webapp/src/components/backstage/channel_selector.tsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ import {GlobeIcon, LockOutlineIcon} from '@mattermost/compass-icons/components';
 import General from 'mattermost-redux/constants/general';
 import {Channel, ChannelMembership} from '@mattermost/types/channels';
 import {Team} from '@mattermost/types/teams';
-import {fetchMyChannelsAndMembers, getChannel} from 'mattermost-redux/actions/channels';
+import {fetchMyChannelsAndMembersREST, getChannel} from 'mattermost-redux/actions/channels';
```
Comment (lieut-data) [inline webapp/src/components/backstage/channel_selector.tsx:13]: Renamed.

File: webapp/src/components/datetime_parsing.ts [hunk 6]
```diff
@@ -101,9 +101,14 @@ const baseUnits = (() => {
...
-        .map((unitDisplay) => new Intl.NumberFormat(locale, {style: 'unit', unit, unitDisplay})
-            .formatToParts(value).find(({type}) => type === 'unit')?.value);
+    return [
+        new Intl.NumberFormat(locale, {style: 'unit', unit, unitDisplay: 'narrow'})
+            .formatToParts(value).find(({type}) => type === 'unit')?.value,
+        new Intl.NumberFormat(locale, {style: 'unit', unit, unitDisplay: 'short'})
+            .formatToParts(value).find(({type}) => type === 'unit')?.value,
+        new Intl.NumberFormat(locale, {style: 'unit', unit, unitDisplay: 'long'})
+            .formatToParts(value).find(({type}) => type === 'unit')?.value,
+    ];
```
Comment (lieut-data) [inline webapp/src/components/datetime_parsing.ts:111]: TypeScript said:
```
src/components/datetime_parsing.ts:105:83 - error TS2322: Type 'string' is not assignable to type '"long" | "short" | "narrow" | undefined'.

105         .map((unitDisplay) => new Intl.NumberFormat(locale, {style: 'unit', unit, unitDisplay})
```

So...

**Comment by nevyangelova** [comment]
🎉 🎉 🎉 🎉 🎉

File: .nvmrc (hunk 2 above)
Comment (calebroseland) [reply .nvmrc:1]: Re: https://github.com/mattermost/mattermost-plugin-playbooks/pull/1608/files#r1041478157, should we also add --legacy-peer-deps like in mattermost-webapp? There are a bunch of warnings now with the react 17<>16 peerDeps and such.

**Review by calebroseland — APPROVED** [review]
LGTM! Just a small bump to package-lock.json to eliminate future/persistent diffs on install.

**Comment by lieut-data** [comment]
Ooh, good catch @calebroseland! Fixed :)

**Pull request merged**